prototype/core/impl/src/main/java/org/eclipse/sensinact/prototype/twin/impl/SensinactDigitalTwinImpl.java

```diff
@@ -147,19 +148,23 @@ private SensinactResourceImpl getResource(Provider provider, String model, Strin
             return null;
         }
 
-        final EStructuralFeature svcFeature = provider.eClass().getEStructuralFeature(service);
+        final EReference svcFeature = (EReference) provider.eClass().getEStructuralFeature(service);
         if (svcFeature == null) {
             return null;
         }
 
-        final EStructuralFeature rcFeature = ((EClass) svcFeature.getEType()).getEStructuralFeature(resource);
+        final EClass serviceEClass = svcFeature.getEReferenceType();
+
+        final ETypedElement rcFeature = serviceEClass.getEOperations().stream()
```

Comment:
Two questions 
1. does this need to be getEAllOperations?
2. Would it make sense to query the structural feature first? Attributes are more common than operations, and the query by name should be faster than searching a list.

Reply:
> does this need to be getEAllOperations?
Right now we don't, as we don't have operations on Objects we inherit from. But this makes sense for the Future. Note: If you ever use getEAll<Something>, keep in mind that you will always will get everything from the EObject level as well. I've added a filter  for that.

> Would it make sense to query the structural feature first? Attributes are more common than operations, and the query by name should be faster than searching a list.

Yes, that makes sense.